### PR TITLE
[multistage][cleanup] make leaf stage also registered with SchedulerService

### PR DESCRIPTION
**Pull request by walterddr**
Summary
===
Allow leaf-stage combine operator to run on the OpChain scheduler, this refactor makes all the execution logic consistent with a single codepath
  - created 2 scheduler service, one for the pull model and one for the push model opChain execution
  - LeafStageTransformOperator is going to be the push/pull model transition point for leaf-stage
      - next PR is going to make LeafStageTransformOperator no-op based
  - allow QueryRunner.processQuery to return PhysicalPlanner or ServerRequestPlanner success/failure result back to broker instead of a blind OK

TODO
===
- separate LeafStageTransformOperator into
    - generic PiplineBreakerOperator (e.g. wait-until-go-ahead-signal operator); and
    - format transformer operator for v1/v2 format compatibility

Design revisit
===
see comments: https://github.com/apache/pinot/pull/10711#issuecomment-1538771214
- revisit yielding strategy, especially for generic yield for opChain due to any particular operator being parked
- revisit mailbox signaling strategy

### CODE REVIEW

**Comment by walterddr** [comment]
after discussion internally we realized several issues. 

1. while receiving mailbox is yielding the opChain by returning a no-op block; sending mailbox is not yielding if it is blocked on a sender: say InMemoryMailbox if the receiving side is not polling it out fast enough then sending opChain will block on only 1 particular mailbox 
2. putting the push-based and pull-based opChains into the same queue exaggerated the above problem b/c of the sending mailbox block
3. this causes other troubles if other operators can potentially cause the OpChain thread to park somewhere that's not returning a no-op block

going back to the drawing board and see what's the best solution. the current solution is to use a cached threadpool for intermediate stage 
PRO: it immediately unblock the distributed deadlock issue. 
CON: under high QPS threads start to thrash. (observed 20-30% overhead in QPS degradation)

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java
Comment (xiangfu0) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java:43]: nit: new line

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java
Comment (xiangfu0) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java:85]: Is it possible that a request is canceled first then OpChain got registered?

**Review by xiangfu0 — APPROVED** [review]

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java
Comment (walterddr) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/ThreadPoolSchedulerService.java]: good point will make it a self-cleaning cache

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/LeafSchedulerService.java
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/LeafSchedulerService.java:82]: q1: cancel is done on a requestID basis or opChain basis
q2: who is keeping track of the opChainID to requestID mapping?
q3: how do we handle out-of-order opChain creation vs. cancellation?

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java:126]: this should be asynchronously submitted and awaits data return for a full opChain style execution. but will follow up once we create a more concrete execution/yielding framework

**Pull request merged**
